docs/howtos/applications/compare_llms.md

```diff
@@ -0,0 +1,203 @@
+testsetgenerator = TestsetGenerator.from_default()
+test_size = 30  # Number of samples to generate
+testset = testsetgenerator.generate(documents, test_size=test_size)
+test_df = testset.to_pandas()
+test_df.head()
+```
+
+<p align="left">
+<img src="../../_static/imgs/compare-llms-testset.png" alt="test-outputs" width="800" height="600" />
+</p>
+
+```{code-block} python
+:caption: collect questions and answers
+test_questions = test_df['question'].values.tolist()
+test_answers = [[item] for item in test_df['answer'].values.tolist()]
+```
+
+
+## Build your RAG
+
+Here I am using llama-index to build a basic RAG pipeline with my documents. The goal here is to collect retrieved contexts and generated answer for each of the test questions from your pipeline. Ragas has integrations with various RAG frameworks which makes evaluating them easier using ragas.
+
+```{note}
+refer to [langchain-tutorial](../integrations/langchain.ipynb) see how to evaluate using langchain
+```
+
+```{code-block} python
+
+import nest_asyncio
+from llama_index import VectorStoreIndex, SimpleDirectoryReader, ServiceContext
+from llama_index.llms import HuggingFaceInferenceAPI
+from llama_index.embeddings import HuggingFaceInferenceAPIEmbedding
+import pandas as pd
+
+nest_asyncio.apply()
+
+
+def build_query_engine(llm):
+    vector_index = VectorStoreIndex.from_documents(
+        documents, service_context=ServiceContext.from_defaults(chunk_size=512, llm=llm),
+        embed_model=HuggingFaceInferenceAPIEmbedding,
+    )
+
+    query_engine = vector_index.as_query_engine(similarity_top_k=2)
+    return query_engine
+```
+
+## Import metrics from ragas
+
+Here we are importing metrics that are required to evaluate retriever component.
+
+```{code-block} python
+from ragas.metrics import (
+    faithfulness,
+    answer_relevancy,
+    answer_correctness,
+)
+
+metrics = [
+    faithfulness,
+    answer_relevancy,
+    answer_correctness,
+]
+```
+
+## Evaluate Zephyr 7B Alpha LLM
+
+For the first llm, I will be using HuggingFace [zephyr-7b-alpha](https://huggingface.co/HuggingFaceH4/zephyr-7b-alpha). I am using HuggingFaceInferenceAPI to generate answers using the model. HuggingFaceInferenceAPI is free to use and token can be setup using [HuggingFaceToken](https://huggingface.co/docs/hub/security-tokens).
+
+```{code-block} python
+# Function to evaluate as Llama index does not support async evaluation for HFInference API
+def generate_responses(query_engine, test_questions, test_answers):
+  responses = [query_engine.query(q) for q in test_questions]
+
+  answers = []
+  contexts = []
+  for r in responses:
+    answers.append(r.response)
+    contexts.append([c.node.get_content() for c in r.source_nodes])
+  dataset_dict = {
+        "question": test_questions,
+        "answer": answers,
+        "contexts": contexts,
+  }
+  if test_answers is not None:
+    dataset_dict["ground_truths"] = test_answers
+  ds = Dataset.from_dict(dataset_dict)
+
+  return ds
+
```

Comment:
Can we have this code in two separate blocks? 
1) build RAG using llama-index
2) Evaluate using Zyphyr